src/main/java/EventsAndCommands/CommandLoader.java

```diff
@@ -2,128 +2,95 @@
 
 import java.lang.reflect.Constructor;
 import java.lang.reflect.InvocationTargetException;
-import java.util.*;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
 import java.util.stream.Collectors;
 
-import com.jagrosh.jdautilities.command.Command;
-
-import net.dv8tion.jda.core.events.message.guild.GuildMessageReceivedEvent;
-import net.dv8tion.jda.core.hooks.ListenerAdapter;
-
-import org.reflections.*;
-import com.mongodb.client.MongoDatabase;
 import com.jagrosh.jdautilities.commons.waiter.EventWaiter;
+import com.mongodb.client.MongoDatabase;
+
+import org.reflections.Reflections;
 
 /**
- * CommandHandler
+ * Autodetects commands that implement a generic type T. Can be used to
+ * dynaically load up all commands without explicitly instantiating them.
+ * Injects needed dependencies into the instances individually. T = The
```

Comment:
should be an `@param <T> The superclass whose subclasses should be loaded`

Reply:
param <T>? I am unaware of that you use param to annotate generic types

Reply:
You do :)
![grafik](https://user-images.githubusercontent.com/20284688/58530977-f8e62300-81e0-11e9-9dd1-5599440b250d.png)